File: python/models/Bitcoin.py
import datetime
import pytz
class Bitcoin:
    
    def __init__(self, unix_timestamp, weighted_price):
        self.original_timestamp = unix_timestamp
        self.timestamp = datetime.datetime.fromtimestamp(float(unix_timestamp), tz= pytz.timezone('UTC')).strftime('%Y-%m-%d')
        self.weighted_price = float(weighted_price)

File: cpp/models/Bitcoin.cpp
//
// Created by Christian C Lei on 2/19/21.
//

#include <chrono>
#include "Bitcoin.h"

using namespace std;

Bitcoin::Bitcoin(const string &originalTimestamp, float weightPrice) : weight_price(weightPrice) {
    char buffer[256];
    long temp = stol(originalTimestamp);
    const time_t old = (time_t)temp;
    struct tm *oldt = gmtime(&old);
    strftime(buffer, sizeof(buffer), "%Y-%m-%d", oldt);
    timestamp = buffer;
}

float Bitcoin::getWeightPrice() const {
    return weight_price;
}

const string &Bitcoin::getTimestamp() const {
    return timestamp;
}